components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java

```diff
@@ -5070,7 +5072,9 @@ private ServiceProviderProperty buildIsAPIBasedAuthenticationEnabledProperty(Ser
         ServiceProviderProperty appleAppId = new ServiceProviderProperty();
         appleAppId.setName(APPLE_APP_ID_PROPERTY_NAME);
         appleAppId.setDisplayName(APPLE_APP_ID_DISPLAY_NAME);
-        appleAppId.setValue(String.valueOf(clientAttestationMetaData.getAppleAppId()));
+        if (clientAttestationMetaData.getAppleAppId() != null) {
```

Comment:
instead of a null check shall we use StringUtils.isNotBlank? in case from the ui if an empty string is passed.

Reply:
Fixed with cd5db0a89aaaa5616efc836d1975e821318dd403